File: 민성/15652.py
# BaekJoon 05/04 2023
# 15652 N과 M(4)
# 12:45 ~ 12:48

N, M = map(int, input().split())

arr = []

def f(index):

    if len(arr) == M:
        print(" ".join(map(str, arr)))
        return
    
    for i in range(index, N+1):
        arr.append(i)
        f(i)
        arr.pop()

f(1)

File: 동규/15652.js
let n, m;
let input = [];
let temp = [];
let result = "";

function recurNum(k, s) {
  if (k === m) {
    ans = temp.join(" ");
    result += ans + "\n";
    return;
  }
  for (let i = s; i < n; i++) {
    temp.push(i + 1);
    recurNum(k + 1, i);
    temp.pop();
  }
}

const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

rl.on("line", (line) => {
  input.push(line);
}).on("close", () => {
  [n, m] = input[0].split(" ").map(Number);

  recurNum(0, 0);
  console.log(result);
});
